aspnet/mvc/overview.rst

```diff
@@ -1,8 +1,313 @@
+
+Domain Model
+############
+
+Many developers choose to encapsulate complex business logic within a *domain model*, which doesn't depend on infrastructure concerns and is easily validated with :doc:`unit tests </testing/index>`. The domain model will often include abstractions and services that allow the Controller to operate at a higher level of abstraction, and keep low-level plumbing code from cluttering the Controller and making it harder to test.
+
+Persistence Model
+#################
+
+Some applications have separate classes that map closely to how data is stored and retrieved from persistence. If you're using Entity Framework, it can usually map your domain model directly to persistence without the need for a separate persistence model.
+
+View Model
+##########
+
+Many developers are familiar with the concept of a ViewModel, especially those who have used application frameworks that use the Model-View-ViewModel pattern. In an MVC web application, a ViewModel is a type that includes just the data a View requires for display (and perhaps sending back to the server). ViewModel types can also simplify `model binding`_. ViewModel types are generally just data containers; any logic they may have should be specific to helping the View render data.
+
+Binding Model
+############
+
+Sometimes it may be worthwhile to create a type specifically for use with `model binding`_. These types are typically just data containers with no behavior.
+
+.. note:: Using a binding model (or viewmodel) is recommended to avoid allowing a malicious user to set properties on the model that are not included in the UI, but are exposed via model binding.
+
+API Model
+#########
+
+If your application exposes an API, the format of the data you expose to clients may be separated from your app's internal domain model by defining custom API model types. This allows you to change your internal model types without impacting clients that may be using your exposed APIs.
+
+Controller Responsibilities
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Controllers are the initial entry point for each request to an MVC app. Their primary responsibility is to work with the model to perform user commands and retrieve data in response to queries. The Controller then performs any necessary mapping from one model type to another (for instance creating a ViewModel needed for a particular View and populating it with results from the Domain Model), and then returns a particular View.
+
+.. note:: It's not uncommon for controllers to become bloated with too many responsibilities. Try to follow the `Single Responsibility Principle <http://deviq.com/single-responsibility-principle/>`_ and push business logic out of the controller and into the domain model whenever possible.
```

Comment:
something like:
note:: Controllers should not be overly complicated with too many responsibilities. To keep controller logic from becoming overly complex, use the  `Single Responsibility Principle <http://deviq.com/single-responsibility-principle/>`_ to push business logic out of the controller and into the domain model.